### DISCUSSION TITLE
Extending the Quantity

### DISCUSSION THREAD

**Discussion by mgberg** [qudt/qudt-public-repo#578] [Ideas]
From the QUDT Schema, Quantity is defined as the following:

> A quantity is the measurement of an observable property of a particular object, event, or physical system. A quantity is always associated with the context of measurement (i.e. the thing measured, the measured value, the accuracy of measurement, etc.) whereas the underlying quantity kind is independent of any particular measurement. Thus, length is a quantity kind while the height of a rocket is a specific quantity of length; its magnitude that may be expressed in meters, feet, inches, etc. Examples of physical quantities include physical constants, such as the speed of light in a vacuum, Planck's constant, the electric permittivity of free space, and the fine structure constant.

The following is an excerpt from [this example in the wiki regarding Quantity Values](https://github.com/qudt/qudt-public-repo/wiki/User-Guide-for-QUDT#7-a-quantity-with-multiple-units):
> The "concept" of the temperature setting is still a single concept, even when its value is represented in multiple units.

Consequently, all the Quantity Values associated with a given Quantity are exactly equivalent, just represented in different units.

However, there are scenarios where it would be useful to have a resource representing the "concept" of a Quantity for which there are multiple Quantity Values (perhaps in the same unit) that are _not_ equivalent. In some scientific and modeling applications, it is the case that a Quantity may be:
 - observed more than once, perhaps using different measurement methods, resulting in slightly different observed values;
 - predicted using one or more physics-based and/or machine learning models, resulting in slightly different predicted values;
 - computed based on other relevant quantities, whose values may or not be estimated;
 - estimated based on the best information available at the given time;
 - or represented by other values via some other relationship than presented above.

Note that a given quantity may have many values for any of the above categories (or others) and any number of values in any given category, and each of those values may have its own uncertainty associated with it. A "best" or "accepted" value may be chosen or aggregated from available observations/predictions/etc. which then could become the "value" of the quantity using the current terminology.

Some examples include:
- The [elastic modulus of aluminum](https://en.wikipedia.org/wiki/Aluminium#:~:text=Young%27s%20modulus,70%C2%A0GPa) is generally accepted to be 70 GPa. A Quantity representing the elastic modulus of aluminum could have a Quantity Value for 70 GPa as well as one for 10153 ksi. However, that value became "accepted" because there have been hundreds of _observations_ of that quantity obtained using rigorous test standards on hundreds of test samples made of aluminum over time. Note that the elastic modulus of each test sample could be represented as its own quantity as well. These hundreds of observations were then aggregated using some accepted method to arrive at 70 GPa. 
- Values for the speed of light (used as an example in the definition of Quantity above) [have gotten more accurate/precise over time](https://en.wikipedia.org/wiki/Speed_of_light#History). Some Greek philosophers thought the speed of light was infinite. The first quantitative estimate of 220000 km/s was made in 1646, it improved to 298000 km/s in 1862, and to 299792.4562±0.0011 km/s (0.006 ppm error) in 1972, then exactly defined as 299792.458 km/s in 1983 (not all estimates are included here for brevity). There have been lots of _estimates_ of the speed of light through history, but likely only one at a time was considered the "accepted" value (or at least some estimations were likely preferred over others by the scientific community at large).
- Like for the speed of light, measurements/estimates of the value of Planck's constant (also used as an example in the definition of Quantity above) [have also gotten more accurate over time](https://www.researchgate.net/profile/Richard-Steiner-3/publication/233948351_History_and_progress_on_accurate_measurements_of_the_Planck_constant/links/00b4952f3a7dea93f1000000/History-and-progress-on-accurate-measurements-of-the-Planck-constant.pdf). The chart below (found in the linked document) highlights the estimates for this quantity over time.
![image](https://user-images.githubusercontent.com/59745812/194067744-206fcb71-5f59-4c8f-87f1-ff87a3800e2e.png)

Note in the above examples that the "concept" represented in each example never changed- the numerical representation of each effectively just became more accurate over time, and it may be useful to know what values went into the determination of the "accepted" value for the quantity as well as the provenance of those values.

Has anyone else given any thought to supporting this kind of capability?

**Comment by dr-shorthair** [comment]
I think what you are focussing in on here is different acts-of-observation, whose results are different estimates of the value or magnitude of a specific quantity. This is the subject of the metrologists bibles [VIM and GUM](https://www.bipm.org/en/committees/jc/jcgm/publications) and also of [ISO 19156:2011](https://www.iso.org/standard/32574.html) (available at no cost as [OGC Topic 20](https://www.ogc.org/standards/om) ) and [W3C SSN Ontology](https://www.w3.org/TR/vocab-ssn/). The process of determining (estimating) quantities is out of scope for QUDT.

**Reply by mgberg** [reply]
Agreed, I'm interested in different acts of observation (or other means of value estimation). I would also agree that the _process for estimating quantities_ is out of scope for QUDT, but I'm wondering about the _expression or representation of estimates of quantities that have already been made_ using some externally defined process.

There already is a lot of overlap between concepts in QUDT and VIM/GUM, and it would be conceivably possible to include more of the concepts defined in VIM/GUM in QUDT.

The W3C SSN ontology is a great example of a way to express how different observations were obtained. Note that SSN conceivably could be compatible with what I was proposing in my original post. A lot of the examples in the SSN documentation generate QUDT Quantity Values for their observations. These Quantity Values could _also_ be the observations from which an aggregate is computed (via some process out of scope for QUDT) like in the example about aluminum above.

**Reply by steveraysteveray** [reply]
I would say that at present, we are providing the qudt:Quantity concept as a class to be used by the application-builder. QUDT does not govern its use, nor do we maintain a vocabulary of Quantities (at least not explicitly - see below). You can, however, create your own subClasses of qudt:Quantity for your own needs, including time-series, estimates of various kinds, etc.

Slight philosophical digression:
If you think about it, qudt:Unit is actually nothing more than a qudt:Quantity that has a well-defined and reproducible set of conditions for its realization and measurement, so my statement that QUDT does not maintain a vocabulary of Quantities is not really true. The same applies to the concept of qudt:PhysicalConstant. We have distinguished these three concepts primarily because they resonate with users as distinct things. qudt:Quantity is thus available for all user-determined contexts. But if a user (you?) were to define a subClassOf qudt:Quantity that met your needs for some kind of definable context that turns out to be very useful and general, perhaps this subClass could become part of the QUDT corpus in the future.

**Reply by mgberg** [reply]
@steveraysteveray Makes sense to me, I'll have to think about it some more.
